pkg/binding/bindingpolicy.go

```diff
@@ -91,7 +91,9 @@ func (c *Controller) handleBindingPolicy(ctx context.Context, objIdentifier util
 			// if the bindingpolicy requires a singleton status, then we should only
 			// have one destination
 			// TODO: this should be removed once we have proper enforcement or error reporting for this
-			clusterSet = pickSingleDestination(clusterSet)
+			if len(clusterSet) > 1 {
```

Comment:
If `len(clusterSet) == 0` then this violates what the user is expected to do; I would log an error at this point. #1950 introduces a better way to report user errors.

Reply:
`len(clusterSet) == 0` may be a transient and legal state during which a user is managing the selection of WEC(s), including change the binding policies' cluster selectors, or changing labels of the WEC(s), or adding/removing WEC(s).

Say the user wants a cluster to report the status. He/she may create the binding policy as the first step, then onboard or label the cluster as the second step. We shouldn't and can't enforce the order of the two steps. Should we consider this as an error?

Reply:
Regardless of how this conversation resolves, there is a bug in `pickSingleDestination` since it would attempt to access the first element in an empty `sets.List`.

@waltforme it's not related to this PR, but if `pickSingleDestination` won't be changed, we should add function documentation on expecting that the parameter `clusterSet` is a none-empty set.

Reply:
@vMaroon Thanks! Not related to this PR but why not just do something here, so I did 7c0c47f3701514e5d5b4d7d942d103eb89a978ea.

Reply:
@waltforme thanks, I meant to handle it in this PR indeed.

Reply:
Indeed the user inconsistency may be transient. Or it may not. At the point in the code that we are discussing here, it is impossible to know. That is why I am dissatisfied with silently ignoring it. #1950 introduces a state-based way to report user errors; that will naturally mean that a transient error will not persistently be reported. Until then, I think the best that can be done is to log a warning.

Reply:
I added logging for the `len(clusterSet) == 0` case in the latest push. The `handleBindingPolicy` func uses non-nil error to signal requeuing, but I think we don't need to requeue for this case. So, I'm not logging an error. I hope there is a `logger.Warning` func but there is no such thing. I am using a `Info` instead.

Reply:
I agree that just logging is fine until we have a more general solution as @MikeSpreitzer pointed out.

Reply:
@waltforme: I still do not see that logging. I looked in https://github.com/waltforme/kubestellar/blob/singleton/pkg/binding/bindingpolicy.go and in https://github.com/kubestellar/kubestellar/commit/df6719e7335b4d776e7adf430c8380745872e948

Reply:
The logging is squashed into https://github.com/kubestellar/kubestellar/pull/2030/commits/5689c99b8d730e85710d1eaa1e9f2c4db648d108. @MikeSpreitzer

Reply:
Oh, sorry, the new logging *is* in the commits where I looked, but not at the place in the logic where I expected it.